src/tokens.js

```diff
@@ -271,101 +272,288 @@ const tokens = {
-   * @param {number} amount Amount of the new token that will be minted
+   * @param {number} amount Amount of the token that will be minted
    * @param {string} pin Pin to generate new addresses, if necessary
+   * @param {boolean} createAnotherMint If should create another mint output after spending this one
+   * @param {boolean} createMelt If should create a melt output (useful when creating a new token)
    *
    * @return {Promise} Promise that resolves when token is minted or an error from the backend arrives
    *
    * @memberof Tokens
    * @inner
    */
-  mintTokens(txId, token, address, amount, pin) {
-    const promise = new Promise((resolve, reject) => {
-      // Authority output token data
-      const tokenData = 0b10000001;
-      // Now we will mint the tokens
-      const newInput = {'tx_id': txId, 'index': 0, 'token': token, 'address': address};
-      // Output1: Mint token amount
-      const tokenOutput1 = {'address': address, 'value': amount, 'tokenData': 1};
-      // Output2: new mint authority
-      const tokenOutput2 = {'address': address, 'value': TOKEN_MINT_MASK, 'tokenData': tokenData};
-      // Output3: new melt authority
-      const tokenOutput3 = {'address': address, 'value': TOKEN_MELT_MASK, 'tokenData': tokenData};
-      // Create new data
-      let newTxData = {'inputs': [newInput], 'outputs': [tokenOutput1, tokenOutput2, tokenOutput3], 'tokens': [token]};
-      try {
-        // Get new data to sign
-        const newDataToSign = transaction.dataToSign(newTxData);
-        // Sign mint tx
-        newTxData = transaction.signTx(newTxData, newDataToSign, pin);
-        // Assemble tx and send to backend
-        transaction.completeTx(newTxData);
-        const newTxBytes = transaction.txToBytes(newTxData);
-        const newTxHex = util.buffer.bufferToHex(newTxBytes);
-        walletApi.sendTokens(newTxHex, (response) => {
-          if (response.success) {
-            resolve();
-          } else {
-            reject(response.message);
+  mintTokens(txID, index, addressSpent, token, address, amount, pin, createAnotherMint, createMelt) {
+    // Get mint data
+    let newTxData = this.getMintData(txID, index, addressSpent, token, address, amount, createAnotherMint, createMelt);
+    return transaction.sendTransaction(newTxData, pin);
+  },
+
+  /**
+   * Generate melt data
+   *
+   * @param {string} txID Hash of the transaction to be used to melt tokens
+   * @param {number} index Index of the output being spent
+   * @param {string} addressSpent Address of the output being spent
+   * @param {string} token Token uid to be melted
+   * @param {number} amount Amount of the token to be melted
+   * @param {boolean} createAnotherMelt If should create another melt output after spending this one
+   *
+   * @return {Object} Melt data {'inputs', 'outputs', 'tokens'}
+   *
+   * @memberof Tokens
+   * @inner
+   */
+  getMeltData(txID, index, addressSpent, token, amount, createAnotherMelt) {
```

Comment:
I would rename it to `createMeltData` (instead of get).

Reply:
Done